ktor-core/src/org/jetbrains/ktor/http/Cookies.kt

```diff
@@ -0,0 +1,150 @@
+package org.jetbrains.ktor.http.cookies
+
+import org.jetbrains.ktor.application.*
+import org.jetbrains.ktor.http.*
+import java.net.*
+import java.time.*
+import java.time.temporal.*
+import java.util.*
+
+public enum class CookieEncoding {
+    RAW, DQUOTES, URI_ENCODING, BASE64_ENCODING
+}
+
+public open class RequestCookies(val request: ApplicationRequest) {
+    private val defaultDecoded: Map<String, String> by lazy { decode(CookieEncoding.URI_ENCODING) }
+
+    public fun get(name: String): String? = defaultDecoded[name]
+    public fun decode(encoding: CookieEncoding): Map<String, String> = request.attributes.getOrPut(CookiesKey(encoding)) {
+        parsedRawCookies.mapValues { decodeCookieValue(it.value, encoding) }
+    }
+
+    public open val parsedRawCookies: Map<String, String> by lazy {
+        request.header("Cookie")?.let { parseClientCookiesHeader(it) } ?: emptyMap()
```

Comment:
Is this the only usage of "request" here? May be we should pass list of cookie headers instead?